draft-ietf-tls-dtls-connection-id.md

```diff
@@ -255,25 +255,37 @@ Other fields are defined in RFC 6347. Note that this specification does
 not make use of the DTLSCompressed structure. 
 
 In addition, the CID value is included in the MAC calculation for the
-DTLS record, as shown below. The MAC algorithm described in Section
-4.1.2.1 of {{RFC6347}} and Section 6.2.3.1 of {{RFC5246}} is extended
-as follows:
+DTLS record layer. At the time of writing ciphers using authenticated 
+encryption with additional data (AEAD) were state-of-the-art. Hence, this 
+specification updates only the additional data calculation defined in 
+Section 6.2.3.3 of {{RFC5246}}, which is re-used by Section
+4.1.2.1 of {{RFC6347}}. 
+
+The additional data calculation is extended as follows:
 
 ~~~~
-      MAC(MAC_write_key, DTLSCompressed.epoch +
-                            DTLSCompressed.sequence_number +
-                            tls12_cid +
-                            DTLSCompressed.version +
-                            cid_length +        // New input
-                            cid +               // New input
-                            DTLSWrappedCompressed.length +
-                            DTLSWrappedCompressed.fragment);
-   where "+" denotes concatenation.
+    additional_data = seq_num + type + version +  
+                      cid + length;
+
+    where "+" denotes concatenation. 
 ~~~~
 
-The `cid_length` field is a single octet containing the length of the
-connection ID.
+seq_num
+: As described in Section 6.2.3.3 of {{RFC5246}} this 64-bit value 
+is formed by concatenating the epoch and the sequence number in the 
+order they appear on the wire.
+
+type
+: This value contains the outer-header content type, i.e. the tls12_cid. 
+
+version
+: This value contains the version number. 
+
+length
+: This value contains the length information in the outer-header. 
 
+cid
+: Value of the connection id. 
```

Comment:
I'd rather say "Value of the *negotiated* connection id" to distinguish it from what is sent on the wire.  And it's maybe worth adding something like "Note that this is empty in case a 0-length CID has been negotiated".

I'd also add an entry for cid_length here, specifying that even if a 0-length CID has been negotiated, and therefore no CID appears on the wire, a cid_length of 0 MUST be added to the AD.

Reply:
I added that text

Reply:
Sorry, if the usage of "empty CID" with the new record type is intended, this is clearly a new use-case and obvious no NAT solution!

And please provide an example, which shows that adding the cid_length make sense!
In other discussion "empty CID in the new record types" are denied.

Reply:
There is no new "payload length obfuscation" use case that is enabled by this change.

I think you are confusing on-the-wire framing with how authentication data is computed.

Hannes's change here is only about MAC computation.  The gist is that 0-length CIDs have 6347 framing but their MAC is computed slightly differently from a "normal" 6347 record.

Reply:
Hm, so want you to ensure, that if a empty CID is used in the Hello Extension, and therefore a RFC 6347 record is used later, the MAC is different compared to a RFC 6347 record, which is used without using a empty cid in the Hello Extension?

Why? What would be the advantage? What are the examples, which show this advantage?

FMPOV: the MAC includes already much keying material, the cid_length willl hardly improve that. From the discussion in #25 two arguments for including the MAC have been named: 

https://github.com/tlswg/dtls-conn-id/issues/25#issuecomment-444890710

"Yes, but it might be true with some extension in the future. Moreover, authenticating the whole header is just good practice."

I simply just don't see the justification, to complicate RFC6347 MAC calculations.

Reply:
> There is no new "payload length obfuscation" use case that is enabled by this change.

In my opinion, this https://github.com/tlswg/dtls-conn-id/issues/22#issuecomment-458758785 indicates, that it is a use-case :-).

Therefore I started #40 .